src/dstore/dstore_internal.h

```diff
@@ -211,6 +214,10 @@ bool dstore_io_vec_invariant(const struct dstore_io_vec *io_vec)
 	 * NOTE: This condition may be changed in future if
 	 * we need to implement Alloc/Free operations.
 	 */
+	if (io_vec->flags & IO_NO_DATA) {
```

Comment:
Let's move it part to the very beginning of this function and explain it:

```c
/* When NO_DATA flag is set, the vector invariant cannot be violated:
 *   the embedded buf has only offset, size -- they are always consistent.
 *   consistency of the extents (when embedded buf is not used) is not checked
 *     anyway, and it is not assumed.
 * Therefore, it is safe to assume that when this flag is set, we have
 * a valid io_vec object.
 */
```

Reply:
Done.